File: py/Euler69.py
def phi(n):
    r = n
    d = 2
    p = n
    while r > 1:
        if r % d == 0:
            p -= int(r/d)
            while r % d == 0:
                r = int(r/d)
        d += 1
    return p

m = (0, 1)
for n in range(2, 1000000):
    p = phi(n)
    m = max(m, (n/p, n))
    if n % 10000 == 0:
        print(n)

print(m)

File: c/Euler69.c
#include <stdio.h>

int phi(int n) {
	int r = n;
	int d = 2;
	int p = n;

	while (r > 1) {
		if (r % d == 0) {
			p -= p / d;
			while (r % d == 0) {
				r = r / d;
			}
		}
		d++;
	}

	return p;
}

int main() {
	double m = 0;
	int mn = 1;
	int i;

	for (i = 2; i < 1000000; i++) {
		int p = phi(i);
		if (m < (double)i / p) {
			m = (double)i / p;
			mn = i;
		}
		if (i % 10000 == 0) {
			printf("%d\n", i);
		}
	}

	printf("%lf %d", m, mn);
}